# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1475263957 18000
#      Fri Sep 30 14:32:37 2016 -0500
# Branch stable
# Node ID d5e775944809586d4753eb7139bdd4bad572adac
# Parent  d14c44923705d1faf325754a4ee8841e87f6b7d7
Use 'color' property when 'markeredgecolor' is 'auto' for gnuplot toolkit (bug #47974).

* __gnuplot_draw_axes__.m (do_linestyle_command): Check if obj.markeredgecolor
  is "auto" and use obj.color if it is to assign an RGB color rather than use
  palette input data.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -1977,19 +1977,24 @@ function [style, ltidx] = do_linestyle_c
         if (! isempty (style{sidx}))
           sidx += 1;
           idx += 1;
         else
           fputs (plot_stream, ";\n");
         endif
         fprintf (plot_stream, "set style line %d default;\n", idx);
         fprintf (plot_stream, "set style line %d", idx);
-        if (isnumeric (obj.markeredgecolor))
+        if (isnumeric (obj.markeredgecolor) || strcmp (obj.markeredgecolor, "auto"))
+          if (isnumeric (obj.markeredgecolor))
+            edgecolor = obj.markeredgecolor;
+          else
+            edgecolor = obj.color;
+          end
           fprintf (plot_stream, " linecolor rgb \"#%02x%02x%02x\"",
-                   round (255*obj.markeredgecolor));
+                   round (255*edgecolor));
         else
           fprintf (plot_stream, " palette");
         endif
         if (! isempty (pt))
           style{sidx} = "points";
           ltidx(sidx) = idx;
           fprintf (plot_stream, " pointtype %s", pt);
         endif
